aee.js

```diff
@@ -82,3 +82,19 @@ Array.prototype.count = function (spec) {
   }
   return result
 }
+
+Array.prototype.max = function (comparer) {
+  const len = this.length
+  if (len === 0) return null
+  let result = this[0]
+  let comparerSpec = comparer
+  if (typeof comparerSpec !== 'function') {
+    comparerSpec = (!isNaN(this[0]) || this[0].constructor === Object)
```

Comment:
i will let you decide what is better for your framework, what it's not for sure is, to access object's properties without testing for null before, ex `this[0].constructor`